src/libhook.js

```diff
@@ -26,4 +26,26 @@
       });
     }
   };
-})();
\ No newline at end of file
+
+  Module['commit_hook'] = function(db, xCommitHook) {
```

Comment:
I think that using the same `pAsyncFlags` for both `update_hook()` and `commit_hook()` is a mistake here. I believe that calling `commit_hook()` will unset any previous `update_hook()` and vice versa.

I believe that using a separate IIFE for each function should work, as they will have their own `pAsyncFlags`.

Reply:
good catch – [c5fd169](https://github.com/rhashimoto/wa-sqlite/pull/225/commits/c5fd16902b8fe013f1a857e8797fb689b63a78f3) adds a failing test which shows that you're right about this, and [7d68b39](https://github.com/rhashimoto/wa-sqlite/pull/225/commits/7d68b39e307a888acd866eeeafcfa522e06f5635) puts the commit hook in its own IIFE, which makes the test pass.

Reply:
Looks good! I just need to verify the binaries and then read through the code one final time.